src/modules/flow/http-server/http-server.c

```diff
@@ -0,0 +1,622 @@
+
+                r = sol_http_server_set_last_modified(server, mdata->path, time(NULL));
+                SOL_INT_CHECK(r, < 0, r);
+            }
+            break;
+        case SOL_HTTP_PARAM_HEADER:
+            if (streq(value->value.key_value.key, HTTP_HEADER_ACCEPT)) {
+                if (strstr(value->value.key_value.value, HTTP_HEADER_CONTENT_TYPE_JSON))
+                    send_json = true;
+            }
+            break;
+        default:
+            break;
+        }
+    }
+
+    if (send_json) {
+        r = snprintf(str, sizeof(str), "{\n\t\"%s\": %s\n}", mdata->path,
+            mdata->value.b == true ? "true" : "false");
+    } else {
+        r = snprintf(str, sizeof(str), "%s", mdata->value.b == true ? "true" : "false");
+    }
+
+    SOL_INT_CHECK_GOTO(r, < 0, end);
+
+    r = sol_http_param_add(&response.param, SOL_HTTP_REQUEST_PARAM_HEADER(
+        HTTP_HEADER_CONTENT_TYPE, (send_json) ? HTTP_HEADER_CONTENT_TYPE_JSON : HTTP_HEADER_CONTENT_TYPE_TEXT));
+    SOL_INT_CHECK_GOTO(r, != true, end);
+
+    r = sol_buffer_set_slice(&response.content, sol_str_slice_from_str(str));
+    SOL_INT_CHECK_GOTO(r, < 0, end);
+
+    r = sol_http_server_send_response(request, &response);
+    SOL_INT_CHECK_GOTO(r, < 0, end);
+
+    if (method == SOL_HTTP_METHOD_POST) {
+        sol_flow_send_boolean_packet(mdata->node,
+            SOL_FLOW_NODE_TYPE_HTTP_SERVER_BOOLEAN__OUT__OUT,
+            mdata->value.b);
+    }
+
+end:
+    sol_buffer_fini(&response.content);
+    sol_http_param_free(&response.param);
+
+    return 0;
+}
+
+static int
+boolean_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id,
+    const struct sol_flow_packet *packet)
+{
+    struct http_data *mdata = data;
+    int r;
+
+    r = sol_flow_packet_get_boolean(packet, &mdata->value.b);
+    SOL_INT_CHECK(r, < 0, r);
+
+    r = sol_http_server_set_last_modified(server, mdata->path, time(NULL));
+    SOL_INT_CHECK(r, < 0, r);
+
+    return 0;
+}
+
+static void
+boolean_close(struct sol_flow_node *node, void *data)
+{
+    struct http_data *mdata = data;
+
+    stop_server(mdata);
+}
+
+static int
+boolean_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    int r;
+    struct http_data *mdata = data;
+    struct sol_flow_node_type_http_server_boolean_options *opts =
+        (struct sol_flow_node_type_http_server_boolean_options *)options;
+
+    r = start_server(mdata, opts->path, boolean_response_cb);
+    SOL_INT_CHECK(r, < 0, r);
+
+    mdata->value.b = opts->value;
+    mdata->node = node;
+
+    return 0;
+}
+
+/* ------------------------------------- string ------------------------------------------- */
+
+static int
+string_response_cb(void *data, struct sol_http_request *request)
+{
+    int r = 0;
+    uint16_t idx;
+    bool send_json = false;
+    enum sol_http_method method;
+    struct http_data *mdata = data;
+    char str[1024] = { 0 };
```

Comment:
That's 1kb of data that's being zeroed out every request. Is it really necessary? Also, for strings, where any arbitrary data can be sent, I'd just use asprintf().